### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by KrisKulbych**

### CODE REVIEW

File: reversed_func.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+def reversed_func(text: str) -> str:
+
```
Comment (DivnychAndrii) [inline reversed_func.py]: Where did docstring go? I loved it

File: reversed_func.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    reversed_text = []
+    letter_list = [letter for word in reversed(text.split()) for letter in word if letter.isalpha()]
```
Comment (DivnychAndrii) [inline reversed_func.py]: I disagree with double for loop in list comprehension. It's too much.  We have to balance between fewer lines of code and the readability of the code. Two and more for-loops in a single list comprehension is too much.

Let's make one regular for-loop that iterates over all words in the initial text and inside make a list comprehension that collects all the letters and a second loop that creates reversed string with symbols and digits.

isalpha is good approach

File: reversed_func.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    for letter in text:
+        if letter.isalpha():
+            reversed_text.append(letter_list.pop())
+        else:
+            reversed_text.append(letter)
+
+    return ''.join(reversed_text)
```
Comment (DivnychAndrii) [inline reversed_func.py]: I would still suggest going word by word and joining them on empty space " ".join(...), looks more readable for me

File: reversed_func.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+cases = [
```
Comment (DivnychAndrii) [inline reversed_func.py]: You are missing if name == main: part

File: reversed_func.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+def reversed_func(text: str) -> str:
```
Comment (DivnychAndrii) [inline reversed_func.py]: Disagree with the name of the function. By forgetting that I know what it does and trying to derive that information from the function's name, I would guess that reversed_func does something to the function so it becomes reversed. Try something different, something more self-speaking

**Review by DivnychAndrii — CHANGES_REQUESTED** [review]

File: reversed_func.py (hunk 1 above)
Comment (KrisKulbych) [reply reversed_func.py]: Sorry, I've lost it accidentally. I fixed and commited it.

File: reversed_func.py (hunk 2 above)
Comment (KrisKulbych) [reply reversed_func.py]: Ok, I got it. I modifyed the list comprehension and added extra for-loop.

File: reversed_func.py (hunk 4 above)
Comment (KrisKulbych) [reply reversed_func.py]: That's again my carelessness, I fixed it.

File: reversed_func.py (hunk 5 above)
Comment (KrisKulbych) [reply reversed_func.py]: I changed the name of the function from reverse_func to reverse_string.
I considered it appropriate to change the names of other variables too. These changes was commited apart.

File: reversed_func.py (hunk 3 above)
Comment (KrisKulbych) [reply reversed_func.py]: Done

**Review by DivnychAndrii — APPROVED** [review]

**Pull request merged**